src/raw-to-zarr-to-bq.py

```diff
@@ -0,0 +1,206 @@
+    first = '{' + '1' + '}'
+    commands = []
+    for DATASET in SPLITTING_DATASETS:
+        command = (
+            f'{PYTHON_PATH} /weather/weather_sp/weather-sp --input-pattern '
+            f' "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Month/{year}/{month}_hres_{DATASET}.grb2" '
+            f'--output-template "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Month/{first}/{zero}.grb2_{typeOfLevel}_{shortName}.grib" '
+            f'--runner DataflowRunner --project {PROJECT} --region {REGION} '
+            f'--temp_location gs://{BUCKET}/tmp --disk_size_gb 3600 '
+            f'--job_name split-{DATASET}-data-{month} --use-local-code '
+        )
+        commands.append(command)
+
+    with ThreadPoolExecutor() as tp:
+        for command in commands:
+            tp.submit(subprocess_run, command)
+
+
+def ingest_data_in_bigquery_dataflow_job(zarr_file: str, table_name: str, region: str,
+                                         zarr_kwargs: str) -> None:
+    """Ingests data from a Zarr file into BigQuery and runs a Dataflow job.
+
+    Args:
+        zarr_file (str): The Zarr file path.
+        table_name (str): The name of the BigQuery table.
+        zarr_kwargs (Any): Additional arguments for the Zarr ingestion.
+
+    Returns:
+        None
+    """
+    if '/ar/' in zarr_file:
+        job_name = zarr_file.split('/')[-1]
+        job_name = os.path.splitext(job_name)[0]
+        job_name = (
+            f"data-ingestion-into-bq-{replace_non_alphanumeric_with_hyphen(job_name)}")
+
+        command = (
+            f"{PYTHON_PATH} /weather/weather_mv/weather-mv bq --uris {zarr_file} "
+            f"--output_table {table_name} --runner DataflowRunner --project {PROJECT} "
+            f"--region {region} --temp_location gs://{BUCKET}/tmp --job_name {job_name} "
+            f"--use-local-code --zarr --disk_size_gb 500 --machine_type n2-highmem-4 "
+            f"--number_of_worker_harness_threads 1 --zarr_kwargs {zarr_kwargs} "
+        )
+
+        subprocess_run(command)
+
+
+def perform_data_operations(z_file: str, table: str, region: str, start_date: str,
+                            end_date: str, init_date: str):
+    # Function to process a single pair of z_file and table
+    try:
+        logger.info(f"Resizing zarr file: {z_file} started.")
+        resize_zarr_target(z_file, end_date, init_date)
+        logger.info(f"Resizing zarr file: {z_file} completed.")
+        logger.info(f"Data ingesting for {z_file} is started.")
+        ingest_data_in_zarr_dataflow_job(z_file, region, start_date, end_date, init_date,
+                                         project=PROJECT, bucket=BUCKET, python_path=PYTHON_PATH)
+        logger.info(f"Data ingesting for {z_file} is completed.")
+        start = f' "start_date": "{start_date}" '
+        end = f'"end_date": "{end_date}" '
+        zarr_kwargs = "'{" + f'{start},{end}' + "}'"
+        # TODO(dabhis): Find a way to ingest AR data into BQ.
```

Comment:
Let's ask @DarshanSP19 to create an issue in weather-tools related to faster ingestion into BQ and mention that bug ID here.